### DISCUSSION TITLE
root task lint (turbo run lint) looks like it invokes turbo and might cause a loop

### DISCUSSION THREAD

**Discussion by tienlx97** [vercel/turborepo#5290] [Help] [answered]
I got this bug when I use `yarn lint`:

root task lint (turbo run lint) looks like it invokes turbo and might cause a loop

- turborepo: "latest"
- netxjs: "latest"

**Comment by thiagodebastos** [comment]
I've been experiencing the same issue since `1.10.3`. It looks like it was fixed in `1.10.4` but now re-introduced in `1.10.6`

**Reply by tienlx97** [reply]
@thiagodebastos me too

**Reply by gabriellet** [reply]
I also see this issue on `1.10.3` and `1.10.6` but not `1.10.4`

**Reply by tienlx97** [reply]
I use     "turbo": "^1.8.8" and it fine, but 1.10.* it fail

**Reply by imMatheus** [reply]
I had the same issue, the boilerplate gave you "turbo": "latest", changing it to "turbo": "1.10.4" fixed it for me

**Reply by kenyipp** [reply]
Downgrade from latest to 1.10.4 work for me

**Reply by d-kja** [reply]
Had the same issue with the latest version, the downgrade worked for me as well.

**Reply by agomesd** [reply]
Downgrading to 1.10.4 doesn't fix it for me

**Reply by anthonyshew** [reply]
I've created an issue for the team to look at based on this report. We appreciate if you could share any reproductions or further information there.

https://github.com/vercel/turbo/issues/6511

**Reply by JBhrayn** [reply]
any updates on this? downgrading to 1.10.4. doesnt fix it for me too

**Comment by anthonyshew** [comment]
Since we don't have any reproductions to look at, I'm going to have to assume that this is not a bug but, rather, that the folks here are running into a small protection that we have in place to make sure you don't recursively call `turbo.`

For instance, if you have:
```
// package.json
{
  "scripts": {
    "lint": "turbo run lint:js lint:css lint:md"
  }
}
```

And try to run `turbo lint`, that would be a `turbo` command inside another `turbo` command, which isn't something we want to support. Instead, you would want to use your package manager to run the `lint` command (e.g. `npm run lint`, `pnpm lint`, etc.).

**Comment by jschuur** [comment]
In case anyone else runs into a 'looks like it invokes turbo and might cause a loop' error, I just got it in a new PNPM monorepo when I initially had a typo in my `pnpm-workspace.yaml` file name. Went away when I used the correct name.

My guess is that the default workspace list that kicks in when you don't have one defined is root.

**Reply by dvelasquez** [reply]
This was my problem. My workspace file was called: `pnpm-workspaces.yaml` when it should be called `pnpm-workspace.yaml`.

After fixing this the problem went away

**Reply by cjkihl** [reply]
This is the correct answer. ✅  If you remove `pnpm-workspace.yaml` turbo throws this error which is miss leading.
Tested on Turbo 2.1.3

## How to reproduce:
1. Download a new turbo repo starter with pnpm `pnpm dlx create-turbo@latest`
2. `Remove the pnpm-workspace.yaml`
3. Run `pnpm run dev`

## Result:
Turbo throws `× root task //#dev (turbo dev) looks like it invokes turbo and might cause a loop`

![image](https://github.com/user-attachments/assets/4cad323c-ab1e-46b1-91c6-87e7e66213ad)

## Suggested result:
It would be great if turbo gives a warning if monorepo configuration could not be found

@anthonyshew  Can we please open this issue again? 🙏

**Reply by anthonyshew** [reply]
Is there a related Issue? We tend to prioritize bugs reported in Issues so if there isn't one yet, it would help us on our end if someone could make one.

I agree this error message should be more informative. Just don't want to forget about here in a Discussion!

**Reply by cjkihl** [reply]
Created one: 
https://github.com/vercel/turborepo/issues/9224

**Reply by anthonyshew** [reply, marked as answer]
Fixing in https://github.com/vercel/turborepo/pull/9650

**Comment by mohammad-amin-hesam** [comment]
in my case I solved this problem with add a pnpm-workspace.yaml at root of project


```
// pnpm-workespace.yaml
packages:
  - "apps/*"
  - "packages/*"
```
